modules/client-common/src/main/java/org/apache/ignite/internal/jdbc/proto/event/JdbcQuerySingleResult.java

```diff
@@ -69,15 +69,30 @@ public JdbcQuerySingleResult(int status, String err) {
         super(status, err);
     }
 
+    /**
+     * Constructor.
+     *
+     * @param hasNext {@code true} if more results are present.
+     * @param updCount Update counter.
+     */
+    public JdbcQuerySingleResult(boolean hasNext, long updCount) {
+        hasResults = hasNext;
+        this.updateCnt = updCount;
+    }
+
     /**
      * Constructor.
      *
      * @param cursorId Cursor ID.
      * @param rowTuples Serialized SQL result rows.
+     * @param columnTypes Ordered list of types of columns in serialized rows.
+     * @param decimalScales Decimal scales in appearance order.
+     * @param isQuery {@code true} if query is SELECT/EXPLAIN.
+     * @param updateCnt Update count.
      * @param last     Flag indicates the query has no unfetched results.
      */
     public JdbcQuerySingleResult(long cursorId, List<BinaryTupleReader> rowTuples, List<ColumnType> columnTypes, int[] decimalScales,
-            boolean last) {
+            boolean last, boolean isQuery, long updateCnt) {
```

Comment:
we already have contractor for DML statements, so let's use that one

Reply:
`JdbcQuerySingleResult(boolean hasNext, long updCount)` is not for DML needs for now.

Reply:
done